Open_Source/Section-5/5.38_Explain_everything_related_to_Flux.md

```diff
@@ -0,0 +1,113 @@
+# Flux
+
+![Flux Image](https://imgur.com/1ZLgq7V.png)
+
+**Flux** is an `architecture` that Facebook uses internally when working with `React`. It is not a framework or a library. It is simply a new kind of **architecture** that complements React and the concept of Unidirectional `Data Flow`.
+
+That said, Facebook does provide a repo that includes a **Dispatcher library**. The `dispatcher` is a sort of global pub/sub handler that broadcasts payloads to registered `callbacks`.
+
+A typical Flux architecture will leverage this Dispatcher library, along with `NodeJS's EventEmitter module` in order to set up an event system that helps manage an applications state.
+
+![Flux Update Cycle](https://imgur.com/k627WyE.png)
+
+
+## Components
+
+- **_Actions_** – Helper methods that facilitate passing data to the Dispatcher 
+- **_Dispatcher_** – Receives actions and broadcasts payloads to registered callbacks 
+- **_Stores_** – Containers for application state & logic that have callbacks registered to the dispatcher 
+- **_Controller Views_** – React Components that grab the state from Stores and pass it down via props to child components
+
+Flux helps to solve some of the difficulty we run into with `unidirectional data` flow when it comes to changing Application State that is higher up the **virtual DOM** than the Components that alter that State themselves.
+
+Controllers do exist in a Flux application, but they are `controller-views`.
+
+`Action creators — dispatcher helper methods` — are used to support a semantic API that describes all changes that are possible in the application. It can be useful to think of them as a fourth part of the Flux update cycle
+
+## How does the API relate to this?
+
+When you are working with data that is coming from (or going to) the outside, I’ve found that using Actions to introduce the data into the Flux Flow, and subsequently Stores, is the most painless way to go about it.
+
+## Action Creators & Actions
+
+**Action Creators** are collections of methods that are called within views (or anywhere else for that matter) to send actions to the Dispatcher. Actions are the actual payloads that are delivered via the dispatcher.
+
+The way Facebook uses them, action type constants are used to define what action should take place, and are sent along with action data. Inside of `registered callbacks`, these actions can now be handled according to their action type, and methods can be `called` with action data as the arguments.
+
+`// ES5`</br>
```

Comment:
you should use triple quote syntax to add this snippet in the markdown file. Please refer the first example  given [here](https://docs.github.com/en/github/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks)